### ISSUE TITLE
Support for object literal property value shorthand

### ISSUE THREAD

**Issue by maryo** [Microsoft/TypeScript#418]
Started here: https://typescript.codeplex.com/workitem/1367

Please consider adding support for object literal property value shorthand, as per the ES6 proposal: http://wiki.ecmascript.org/doku.php?id=harmony:object_literals#object_literal_property_value_shorthand. This would allow for the following:

```typescript
function foo(a: number) {
  var b = 123;
  return { a, b };  // instead of { a: a, b: b };
}
```

**Comment by Arnavion** [comment]
Fixed by #1127 and #1346

**Comment by basarat** [comment]
> In my codebase this feature would very useful, because I'm often creating "smaller" object from "bigger" one, so for loop on "smaller" object enumerates just the properties I need.

You can use destructuring with object literal property value: 

```ts
var bigObject = {
    A: "1",
    B: "2",
    C: "3"
};

var {A,B} = bigObject,
    smallObject = {A,B};
```

:rotating_light: untested code
